src/dialog/src/Dialog.js

```diff
@@ -299,7 +331,11 @@ class Dialog extends React.Component {
             marginY={topOffsetWithUnit}
             display="flex"
             flexDirection="column"
-            css={animationStyles}
+            css={animationStyles(
```

Comment:
I wonder if we could make use of glamor's `compose` instead? That way consumers can literally put whatever styles and animations they want – not restricted to keyframes.

Reply:
@mshwery I'm not sure if I understood what you meant; since the consumers are able to put whatever styles they want on the `containerProps`; even overwrite the `CSS` property.

Can you please explain to me a little bit more your thoughts?

Reply:
Great point – so this is mostly just a convenience method to expose glamor's `keyframes` in a simplified way to consumers?

Why not override with the `css` prop like you suggested?

Reply:
If we would do this overwriting the CSS property with the `containerProps`, we would need to implement all the following variables/functions _(`animationStyles`, `generateOpenAnimation` and `generateCloseAnimation`)_ on the client side. So I believe that, in this way (I implemented on this PR), we're going to have a cleaner API.

Reply:
@mshwery Can you please give me your thoughts?

Reply:
I think your API for this component is pretty sound, and it seems like a reasonable benefit/simplification.

While I think it's a good addition, I'm concerned that this is an ad-hoc feature that only benefits this component. Something like this should probably be more "baked-in" to the design of the library in a consistent way across all/many components – any that have animations or css to override...

I'm also not sure that the library needs to solve this immediately, because as you said – you can achieve this through props today, although it's a bit more verbose. One could write a wrapping component that provides this interface, if you find yourself doing it often.

I'll have to share with the rest of the team and would love feedback from others in the community. :) 

Can we put this on hold until we get that feedback? Also, I would be open to exploring a more holistic approach that solves the problem across multiple components.

Reply:
@mshwery I agree with you. This change needs to be done in a general way; so all the components can take advantage of it. Otherwise, we're going to have components with different behavior from each other.

Let's put it on hold, as you said. If you need anything in this meanwhile, please let me know. I'll try to develop something more generic for it